lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,180 @@
+export { Props };
+
+import '../NxFileUpload.scss';
+
+function SelectedFileWrapper({file, onDismiss: onDismissProp}: MultiSelectedFileProps) {
+  const fileRef = useRef<HTMLSpanElement>(null);
+
+  function onDismiss() {
+    onDismissProp(file);
+  }
+
+  useEffect(function() {
+    if (fileRef.current) {
+      fileRef.current.scrollIntoView({block: 'nearest'});
+    }
+  }, []);
+
+  return (
+    <SelectedFile file={file} onDismiss={onDismiss} ref={fileRef}/>
+  );
+}
+
+const NxMultiFileUpload = forwardRef<HTMLDivElement, Props>(function NxMultiFileUpload(props, ref) {
+  const {
+        className: classNameProp,
+        onChange: onChangeProp,
+        files,
+        isRequired,
+        isPristine,
+        id,
+        disabled,
+        ...attrs
+      } = props,
+      isFileSelected = !!files?.length,
+      { showValidationErrors: formShowValidationErrors } = useContext(FormAriaContext),
+      showValidationErrors = formShowValidationErrors || !isPristine,
+      showError = isRequired && showValidationErrors && !isFileSelected,
+      className = classnames('nx-multi-file-upload', classNameProp),
+      noFileMessageClassName = classnames('nx-multi-file-upload__no-file-message', {
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error');
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    onChangeProp(dataTransferObject.files);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+
+    if (files) {
+      const filesArray = Array.from(files);
+      filesArray.map((f) => {
+        if (f === fileObj) {
+          const idx = filesArray.indexOf(f);
+          if (idx === files.length - 1) {
+            // if the selected file is the last in the container, set the focus to the next last dismiss button
+            // if selected file is only file, set focus on the input
+            closeBtns.length === 1 ? inputRef.current?.focus() : closeBtns[idx - 1].focus();
+          }
+          else {
+            // need to explicity set focus on next closeBtn for Safari
+            closeBtns[idx].focus();
```

Comment:
Safari doesn't consider buttons to be focusable, and we generally allow it to go with that as that's what its users would apparently expect.  So I wouldn't handle this case specially.

Reply:
> Safari doesn't consider buttons to be focusable, and we generally allow it to go with that as that's what its users would apparently expect. So I wouldn't handle this case specially.

That makes sense. If I remove the handling of setting the focus to the next close button in the `else` statement, then Safari doesn't set focus to the next following dismiss button, BUT it will still set focus to the next-last dismiss button when the last dismiss button is clicked, as was design's request. Not ideal, but is this ok to leave?

Reply:
> That makes sense. If I remove the handling of setting the focus to the next close button in the else statement (since this isn't needed for Chrome or Firefox), then Safari doesn't set focus to the next following dismiss button, BUT it will still set focus to the next-last dismiss button when the last dismiss button is clicked, as was design's request. Not great, but is this ok to leave?

As discussed on Slack part of the problem here is that the React keys that we are using aren't very good.  When you click one dismiss button, you expect that button to be removed from the DOM but what actually happens is the last button is removed and the rest just change which file they pertain to.  This makes focus management confusing.  See https://github.com/sonatype/sonatype-react-shared-components/compare/RSC-1158_Multiple_File_Upload_component...RSC-1158_Multiple_File_Upload_component_ross for a way to use keys that are stable per-file.  From there we can work on improving the focus management code.

Reply:
> As discussed on Slack part of the problem here is that the React keys that we are using aren't very good. When you click one dismiss button, you expect that button to be removed from the DOM but what actually happens is the last button is removed and the rest just change which file they pertain to. This makes focus management confusing. See [RSC-1158_Multiple_File_Upload_component...RSC-1158_Multiple_File_Upload_component_ross](https://github.com/sonatype/sonatype-react-shared-components/compare/RSC-1158_Multiple_File_Upload_component...RSC-1158_Multiple_File_Upload_component_ross) for a way to use keys that are stable per-file. From there we can work on improving the focus management code.

Thanks for looking into this and implementing a correction. I merged your branch here [f75a21f](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/f75a21ff2dc54866d555877253059e026139351f)
Then fixed focusing issues here: [f54a520](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/f54a52087f5c4794ea2ceff284421263f955cebd). I have reached out to Jason already about a scrolling jump that now happens when the next dismiss button becomes focused. I've given him two options (with or without `preventScroll=true` on `closeBtns[idx + 1].focus()`) and waiting on reply.

Reply:
> I have reached out to Jason already about a scrolling jump that now happens when the next dismiss button becomes focused. I've given him two options (with or without preventScroll=true on closeBtns[idx + 1].focus()) and waiting on reply.

I think preventing the scroll makes sense.  Assuming the surrounding code handles the `FileList` change correctly, the newly focused button should always end up in exactly the same spot as the button that was just clicked, which means it should end up at least partially in view (unless the user did something like tabbed to it, then scrolled, then pressed Enter, but that's on them. The newly focused button will be in view to exactly the same extent as the button they just pressed).

Reply:
> I think preventing the scroll makes sense. Assuming the surrounding code handles the `FileList` change correctly, the newly focused button should always end up in exactly the same spot as the button that was just clicked, which means it should end up at least partially in view (unless the user did something like tabbed to it, then scrolled, then pressed Enter, but that's on them. The newly focused button will be in view to exactly the same extent as the button they just pressed).

I completely agree as well. Double checked with Jason, and added` preventScroll` here: [db00c5f](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/db00c5fd1c3f17cba05d5a1b68e4520b50e0b963)